proto/client/gClient.go

```diff
@@ -171,18 +172,22 @@ func (confClient *ConfigClient) subscribeToConfigPod(commChan chan *protos.Netwo
 					logger.GrpcLog.Errorf("Failed to subscribe: %v", err)
 					time.Sleep(time.Second * 5)
 					// Retry on failure
+					maxRetryCount--
 					continue
 				}
 			} else if status == connectivity.Idle {
 				logger.GrpcLog.Errorf("Connectivity status idle, trying to connect again")
 				time.Sleep(time.Second * 5)
+				maxRetryCount--
 				continue
 			} else {
 				logger.GrpcLog.Errorf("Connectivity status not ready")
 				time.Sleep(time.Second * 5)
+				maxRetryCount--
 				continue
 			}
 		}
+
 		rsp, err := stream.Recv()
```

Comment:
First - Could you please confirm that in your case SMF crashes at line 191 after 10 retry....
Second - I am not as such in favor of restarting all network functions.. Do you think we can just reinitialize the grpc and connect & get the configuration?

Reply:
@gatici, I am seeing the same thing as @thakurajayL indicated here.
For me the `smf` crashes after the `webui` pod gets restarted. Below is a snippet of the log/issue:
```
2024-09-28T02:18:44Z [ERRO][Config5g][GRPC] Failed to receive message: rpc error: code = Unavailable desc = error reading from server: EOF
2024-09-28T02:18:49Z [ERRO][Config5g][GRPC] Connectivity status idle, trying to connect again
github.com/omec-project/config5g/proto/client.(*ConfigClient).subscribeToConfigPod(0xc00057adc0, 0xc0000db3b0)
        /go/src/smf/vendor/github.com/omec-project/config5g/proto/client/gClient.go:191 +0x2d3 fp=0xc00088ffc0 sp=0xc00088fef0 pc=0xb5df33
github.com/omec-project/config5g/proto/client.(*ConfigClient).PublishOnConfigChange.gowrap1()
        /go/src/smf/vendor/github.com/omec-project/config5g/proto/client/gClient.go:79 +0x25 fp=0xc00088ffe0 sp=0xc00088ffc0 pc=0xb5d425
created by github.com/omec-project/config5g/proto/client.(*ConfigClient).PublishOnConfigChange in goroutine 1
        /go/src/smf/vendor/github.com/omec-project/config5g/proto/client/gClient.go:79 +0xa7
```